src/vea/vea_free.c

```diff
@@ -15,33 +15,133 @@ enum vea_free_type {
 	VEA_TYPE_PERSIST,
 };
 
+int
+free_type(struct vea_space_info *vsi, uint64_t blk_off, uint32_t blk_cnt, bool transient,
+	  uint16_t *class)
+{
+	int			 type = VEA_FREE_ENTRY_BITMAP;
+	struct vea_free_bitmap	*found, *last_found = NULL;
+	daos_handle_t		 btr_hdl;
+	d_iov_t			 key_in, key_out, val;
+	uint64_t		 found_end, vfe_end;
+	int			 rc, opc = BTR_PROBE_LE;
+	bool			 cross_bitmap = false;
+
+	*class = 0;
+	if (blk_cnt > vsi->vsi_class.vfc_large_thresh) {
```

Comment:
All allocations greater than VEA_MAX_BITMAP_CLASS must be extent alloc.

Reply:
Yes, if merging don't cross bitmap.